File: api/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{


    public class Order : IDBEntity<Order>
    {
        [Key]
        public int OrderId { get; set; }

        [Required]
        public OrderStatusEnum Status { get; set; } = OrderStatusEnum.UNPAID;

        [Required]
        public DateTime TimeIn { get; set; }
        public DateTime TimeOut { get; set; }

        [Required]
        [ForeignKey("Customer")]
        public ulong CustomerId { get; set; }
        [Required]
        [ForeignKey("Vehicle")]
        public int VehicleId { get; set; }
        [ForeignKey("Invoice")]
        public int InvoiceId { get; set; }
        
        public string? EstimateNumber { get; set; }
        public string? ScopeOfWork { get; set; }
        public float Expenses { get; set; }

        public override void AssignTo(Order other)
        {
            Status = other.Status;
            TimeIn = other.TimeIn;
            TimeOut = other.TimeOut;
            CustomerId = other.CustomerId;
            VehicleId = other.VehicleId;
            InvoiceId = other.InvoiceId;
            EstimateNumber = other.EstimateNumber;
            ScopeOfWork = other.ScopeOfWork;
            Expenses = other.Expenses;
        }
    }
}

File: api/models/order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order = void 0;
const crypto_1 = require("crypto");
const mongoose_1 = require("mongoose");
const StatusEnum = [
    "PAID",
    "UNPAID",
    "OK",
    "PENDING",
    "WITH BALANCE",
    "QUOTE OR CHECK",
    "FOR LOA OR INVOICE"
];
const TypeEnum = [
    "PERSONAL",
    "WALK IN",
    "FLEET",
    "INSURANCE"
];
const DEFAULT_STATUS = "UNPAID";
const DEFAULT_TYPE = "PERSONAL";
const OrderSchema = new mongoose_1.default.Schema({
    id: { type: String, default: (0, crypto_1.randomUUID)() },
    status: {
        type: String,
        enum: StatusEnum,
        default: DEFAULT_STATUS
    },
    timeIn: { type: Date },
    timeOut: { type: Date },
    customer: String,
    type: {
        type: String,
        enum: TypeEnum,
        default: DEFAULT_TYPE
    },
    company: String,
    vehicle: String,
    invoice: {
        id: { type: String, default: (0, crypto_1.randomUUID)() },
        amount: mongoose_1.default.Types.Decimal128,
        deductible: mongoose_1.default.Types.Decimal128,
        agentFirstName: String,
        agentLastName: String,
        datePaid: Date,
        agentComission: mongoose_1.default.Types.Decimal128
    },
    estimateNumber: String,
    scopeOfWork: String,
    expenses: [{
            id: { type: String, default: (0, crypto_1.randomUUID)() },
            dateRecorded: Date,
            description: String,
            amount: mongoose_1.default.Types.Decimal128,
        }]
});
exports.Order = mongoose_1.default.model('Order', OrderSchema);
